test/apm_repo.js

```diff
@@ -76,20 +75,28 @@ contract('Repo', accounts => {
 
         it('fails when changing contract address in non major version', async () => {
             return assertRevert(async () => {
-                await repo.newVersion([1, 1, 0], accounts[2], initialContent)
+                return await repo.newVersion([1, 1, 0], accounts[2], initialContent)
             })
         })
 
         it('fails when version bump is invalid', async () => {
             return assertRevert(async () => {
-                await repo.newVersion([1, 2, 0], initialCode, initialContent)
+                return await repo.newVersion([1, 2, 0], initialCode, initialContent)
             })
         })
 
         it('fails if requesting version 0', async () => {
-            return assertRevert(async () => {
-                await repo.getByVersionId(0)
-            })
+            try {
+                let result = await repo.getByVersionId(0)
+            } catch (e) {
+                if (e.message.search('base 16 number') > -1) {
+                    // geth and parity will return values for the repo version which web3 chokes on
+                    // and determines that they cannot be converted to uint16 so we need to catch this error
+                    return true
+                } else {
+                    return assert.isAbove(e.message.search('revert'), -1, 'should have failed with revert')
+                }
+            }
```

Comment:
Somewhat confused by this—`geth` and `parity` don't fail on this but rather, return values? The error that occurs (`e.message.search('base 16 number')`) is from web3 failing to convert the return value?

Sounds nuts. What repo version do they return?

In any case, it might be nice to wrap this so we can still use the generic `assertRevert`:

```js
return assertRevert(async () => {
  try {
    await repo.getByVersionId(0)
  } catch (e) {
    if (e.message.search('base 16 number') > -1) {
      // geth and parity will return values for the repo version which web3 chokes on
      // and determines that they cannot be converted to uint16 so we need to catch this error
        throw new Error('mock revert')
      } else {
        throw e
      }
    }
  }
})
```